### PR TITLE
Emb tauid sf

### PR DESCRIPTION
**Pull request by conformist89**
Tau ID UL 2018

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -138,3 +138,18 @@ dmypy.json
...
 *.html
+
+#txt
+*.txt
+
+#out
+*.out
+
+#JSON
+*.json
```
Comment (harrypuuter) [inline .gitignore:149]: I would not add json to the gitignore

File: config/shapes/channel_selection.py [hunk 2]
```diff
@@ -1,26 +1,116 @@
...
-                    ("id_tau_vsMu_Tight_2>0.5", "againstMuonDiscriminator"),
-                    ("id_tau_vsEle_VVLoose_2>0.5", "againstElectronDiscriminator"),
-                    ("id_tau_vsJet_Tight_2>0.5", "tau_iso"),
-                    ("iso_1<0.15", "muon_iso"),
-                    ("mt_1 < 70", "mt_cut"),
-                ]
-            )
+            if wp == "vvtight":
+                cuts.extend(
+                    [
```
Comment (harrypuuter) [inline config/shapes/channel_selection.py:34]: cant we put this if statements into a single function as well (using the wp dict on top with something like 

```py

(f"id_tau_vsJet_{vs_ele_dict[wp]}_2>0.5", "tau_iso"),
```

File: config/shapes/channel_selection.py [hunk 3]
```diff
@@ -161,16 +332,43 @@ def channel_selection(channel, era, special=None):
...
-                ("pzetamissvis > -25", "pzetamissvis"),
-                ("mt_1 < 60", "mt_1"),
-            ]
-        )
+        if wp == "tight":
+            cuts.extend(
+                [
+                    ("id_tau_vsMu_Tight_2>0.5", "againstMuonDiscriminator"),
+                    ("id_tau_vsEle_VLoose_2>0.5", "againstElectronDiscriminator"),
+                    ("id_tau_vsJet_Tight_2>0.5", "tau_iso"),
```
Comment (harrypuuter) [inline config/shapes/channel_selection.py:340]: see comment above

File: embeddded_tauID_ul2018.sh [hunk 4]
```diff
@@ -280,46 +288,74 @@ if [[ $MODE == "PLOT-POSTFIT" ]]; then
...
+        python3 plotting/plot_shapes_tauID_postfit.py -l --era ${ERA} --input ${FILE} --channel mm --embedding --single-category 100 --categories "None" -o output/postfitplots_muemb_${TAG}/${WP}
     done
     exit 0
 fi
 if [[ $MODE == "PLOT-SF" ]]; then
-    source utils/setup_root.sh
-    echo python3 plotting/plot_TauID_sf.py --input output/$datacard_output/ --output output/postfitplots_muemb/ --wp ${WP}
-    python3 plotting/plot_TauID_sf.py --input output/$datacard_output/ --output output/postfitplots_muemb/ --wp ${WP}
+    # source utils/setup_root.sh
+    source /work/olavoryk/source_files/setup-centOS7-gcc10.sh
```
Comment (harrypuuter) [inline embeddded_tauID_ul2018.sh:304]: this should be a relative path using scripts in the repo

File: impacts_all.sh
Comment (harrypuuter) [inline impacts_all.sh:1]: I suggest to move this scipt, and the `log_like_scan.sh` and `log_scan_freeze_np.sh` into a seperate folder called `combine_scripts` to keep everything a bit more organized

File: log_scan_freeze_np.sh
Comment (harrypuuter) [inline log_scan_freeze_np.sh:1]: see comment above

**Comment by conformist89** [comment]
Just updated with respect to the comments above

**Review by harrypuuter — APPROVED** [review]

**Comment by harrypuuter** [comment]
thank you !

**Pull request merged**